### ISSUE TITLE
Override compiler version from a snapshot

### ISSUE THREAD

**Issue by snoyberg** [commercialhaskell/stack#836]
Motivation: wanting to use a different point release, or test out a new GHC HEAD image. Perhaps the syntax could be:

resolver:
    snapshot: lts-3.2
    ghc: 7.10.3

Splitting off from #816

**Comment by phadej** [comment]
Isn't this possible by setting `$PATH` *right*, and using `--skip-ghc-check`?

**Comment by snoyberg** [comment]
Yes, but this would have the advantage that it would be explicit in the
config file, setup would work, and stack could prevent you from
accidentally using the wrong GHC.

**Comment by phadej** [comment]
:+1:

**Comment by phadej** [comment]
Should one consider also command line option? Example usage:
```
stack --resolver-ghc=7.10.3 --stack-yaml=stack-lts-3.yaml build --test
```

and similarly `--resolver-snapshot` (or just `--snapshot?`) ?

**Comment by snoyberg** [comment]
I'd rather not tbh. It's going to be complicated enough to figure out a
good syntax for the yaml, I doubt we'll come up with something good on the
command line.

**Comment by snoyberg** [comment]
Closed in favor of #863
